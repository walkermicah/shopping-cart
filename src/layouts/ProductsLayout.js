import { Outlet, Link } from 'react-router-dom';
import '../styles/layouts/ProductsLayout.scss';

function ProductsLayout() {
  return (
    <div className="ProductsLayout">
      <nav>
        <ul>
          <li>
            <Link to="/products">All</Link>
          </li>

          <li>
            <Link to="produce">Produce</Link>
          </li>

          <li>
            <Link to="bakery">Bakery</Link>
          </li>

          <li>
            <Link to="dairy">Dairy/Eggs</Link>
          </li>
        </ul>
      </nav>
      <Outlet />
    </div>
  );
}

export default ProductsLayout;
