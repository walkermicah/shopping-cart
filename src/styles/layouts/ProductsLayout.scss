.ProductsLayout {
  height: 100%;

  nav {
    padding: 2rem;

    ul {
      display: flex;
      gap: 2rem;
      list-style: none;
    }

    li {
      padding: 0.5rem 2rem;
      border-radius: 15px;
      background-color: black;
      transition: transform 0.2s;

      &:hover {
        transform: translateY(-1px);
      }

      a {
        font-size: 1.75rem;
        text-decoration: none;
        color: white;
      }
    }
  }
}

/* 400px  */
@media only screen and (max-width: 25em) {
  .ProductsLayout nav {
    padding: 2rem 0.5rem;

    ul {
      gap: 1rem;
    }
  }
}
