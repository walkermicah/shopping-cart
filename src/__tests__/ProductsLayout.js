import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import ProductsLayout from '../layouts/ProductsLayout';

describe('Products Layout Component', () => {
  it('Renders products layout navigation', () => {
    const { container } = render(<ProductsLayout />, {
      wrapper: BrowserRouter,
    });
    expect(container).toMatchSnapshot();
  });
});
