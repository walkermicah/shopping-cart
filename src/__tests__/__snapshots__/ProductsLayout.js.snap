// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products Layout Component Renders products layout navigation 1`] = `
<div>
  <div
    class="ProductsLayout"
  >
    <nav>
      <ul>
        <li>
          <a
            href="/products"
          >
            All
          </a>
        </li>
        <li>
          <a
            href="/produce"
          >
            Produce
          </a>
        </li>
        <li>
          <a
            href="/bakery"
          >
            Bakery
          </a>
        </li>
        <li>
          <a
            href="/dairy"
          >
            Dairy/Eggs
          </a>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
